docs/reference/index-modules/slowlog.asciidoc

```diff
@@ -58,29 +58,85 @@ The search slow log file is configured in the `log4j2.properties` file.
 [discrete]
 ==== Identifying search slow log origin
 
-It is often useful to identify what triggered a slow running query. If a call was initiated with an `X-Opaque-ID` header, then the user ID
-is included in Search Slow logs as an additional **id** field
+It is often useful to identify what triggered a slow running query.
+To include information about the user that triggered a slow search,
+use the `index.search.slowlog.include.user` setting.
+
+[source,console]
+--------------------------------------------------
+PUT /my-index-000001/_settings
+{
+  "index.search.slowlog.include.user": true
+}
+--------------------------------------------------
+// TEST[setup:my_index]
+
+This will result in user information being included in the slow log.
 
 [source,js]
 ---------------------------
 {
-  "type": "index_search_slowlog",
-  "timestamp": "2030-08-30T11:59:37,786+02:00",
-  "level": "WARN",
-  "component": "i.s.s.query",
-  "cluster.name": "distribution_run",
-  "node.name": "node-0",
-  "message": "[index6][0]",
-  "took": "78.4micros",
-  "took_millis": "0",
-  "total_hits": "0 hits",
-  "stats": "[]",
-  "search_type": "QUERY_THEN_FETCH",
-  "total_shards": "1",
-  "source": "{\"query\":{\"match_all\":{\"boost\":1.0}}}",
-  "id": "MY_USER_ID",
-  "cluster.uuid": "Aq-c-PAeQiK3tfBYtig9Bw",
-  "node.id": "D7fUYfnfTLa2D7y-xw6tZg"
+  "@timestamp": "2024-02-21T12:42:37.255Z",
+  "log.level": "WARN",
+  "auth.type": "REALM",
+  "elasticsearch.slowlog.id": null,
```

Comment:
This is currently included as `null` (this is the `X-Opaque-ID` value) I wonder if we should change existing behaviour to exclude it instead?

Reply:
The reason I was thinking it might make sense is because we exclude other missing fields with the new code, for example `apikey.id` and `apikey.name`. The use case for `X-Opaque-ID` is a little different though, it's probably useful for someone debugging to see that the header they're providing is not being received for some reason. Let's keep it as is. 

I agree, the duplication doesn't add much. I've removed it.